### PR TITLE
Feature: Add admin API endpoint for platform registration

### PR DESCRIPTION
**Pull request by kengio**

### CODE REVIEW

File: skel/handlers/middlewares.go [hunk 1]
```diff
@@ -12,3 +16,73 @@ func KaigaraConfigMiddleware(config *kaigara.KaigaraConfig) gin.HandlerFunc {
...
+			c.JSON(http.StatusUnauthorized, gin.H{"error": "Unauthorized"})
+			return
+		}
+
+		c.Next()
+	}
+}
+
+func isAdminRole(role string) bool {
+	var roles = []string{"superadmin", "admin", "accountant", "compliance", "support", "technical"}
```
Comment (calj) [inline skel/handlers/middlewares.go:79]: we need to think about this one, maybe this list is good enough as a first filter for the /api/v2/admin/*
then for the registration endpoint we need to strictly limit to superadmin only.

Comment (kengio) [reply skel/handlers/middlewares.go:79]: I will keep this function and add more check for only "superadmin" role in CreatePlatform handler.

Is it ok?

File: skel/handlers/middlewares.go [hunk 2]
```diff
@@ -12,3 +17,109 @@ func VaultConfigMiddleware(conf *sonic.VaultConfig) gin.HandlerFunc {
...
+
+		// Get global vault service
+		vaultService, err := GetGlobalVaultService(c)
+		if err != nil {
+			c.Abort()
+			c.JSON(http.StatusNotFound, gin.H{"error": err.Error()})
+			return
+		}
+
+		key := "sonic_public_key"
```
Comment (calj) [inline skel/handlers/middlewares.go:56]: There is no reason to put this logic in the AuthMiddleware.
It doesn't have anything to do with the Auth!
It only relates to the registration endpoint, so it should in the handler directly.
Also we said we need to retrieve the 3 public keys: peatio, barong and sonic.

Comment (calj) [reply skel/handlers/middlewares.go:56]: Finally we will inject those keys into the environment, so we can skip the vault part here and directly read them from env variables, let's say named:
- PEATIO_PUBLIC_KEY
- BARONG_PUBLIC_KEY
- SONIC_PUBLIC_KEY

Comment (kengio) [reply skel/handlers/middlewares.go]: Ok, I can get all public keys from ENV directly. But

1. Which public key used to verify JWT in Sonic’s admin endpoint? Do I need another ENV name JWT_PUBLIC_KEY for JWT verification only?
2. How I know which public keys to be used during platform registration (peatio, barong or sonic)?

File: skel/handlers/middlewares.go [hunk 3]
```diff
@@ -12,3 +17,82 @@ func VaultConfigMiddleware(conf *sonic.VaultConfig) gin.HandlerFunc {
...
+	return func(c *gin.Context) {
+		// Get bearer token from header
+		authHeader := strings.Split(c.GetHeader("Authorization"), "Bearer ")
+		if len(authHeader) != 2 {
+			c.Abort()
+			c.JSON(http.StatusBadRequest, gin.H{"error": "Authorization header not found"})
+			return
+		}
+
+		// Load public key
```
Comment (calj) [inline skel/handlers/middlewares.go:48]: Can you try to move the keyStore loading before the return line 39, so we do that only once, not at every request.

File: skel/handlers/middlewares.go (hunk 2 above)
Comment (kengio) [reply skel/handlers/middlewares.go]: Fixed.

**Pull request merged**